core/store/src/flat/inlining_migration.rs

```diff
@@ -378,8 +386,16 @@ mod tests {
         ];
         populate_flat_store(&store, shard_uid, &values);
 
-        let flat_storage_manager =
-            FlatStorageManager::test(store.clone(), &vec![shard_uid], CryptoHash::default());
+        let flat_storage_manager = FlatStorageManager::new(store.clone());
+        let mut store_update = store.store_update();
+        flat_storage_manager.set_flat_storage_for_genesis(
+            &mut store_update,
+            shard_uid,
+            &CryptoHash::default(),
+            0,
+        );
+        store_update.commit().unwrap();
+        flat_storage_manager.create_flat_storage_for_shard(shard_uid).unwrap();
```

Comment:
seems redundant as we are seeing exactly same block three times. Anything we can do about it?

Reply:
+1, please create a helper function for tests in this file, something like `fn create_flat_storage_for_genesis(store: Store, shard_uid: ShardUId) -> FlatStorageManager`